tests/runtest.py

```diff
@@ -280,6 +291,12 @@ def setup(self):
     def runcmd(self):
         """ This function returns (shell) command that runs the test.
             A test case can extend this to setup a complex configuration.  """
+
+        if self.p_flag == '-P .' and len(self.p_libs) > 0:
+            self.p_flag = ''
+            for lib in self.p_libs:
+                self.p_flag += '-P .@%s ' % lib
```

Comment:
What about this?

```python
if self.p_flags != '':
    for lib in self.p_libs:
        self.p_flags += ' -P .@%s' % ilb
```

Reply:
It doesn't work in `t077_lib_replay_T` because the expected result doesn't contain the tracing result of the main executable binary so we should remove `-P .` in this case.

If I change the code as follows.
```py
        if self.p_flag != '':
            self.p_flag = ''
            for lib in self.p_libs:
                self.p_flag += ' -P .@%s' % lib
```
Then it will break 198 lib_arg_float.
```
$ ./runtest.py -vde -O0 -c gcc 198
Start 1 tests without worker pool
Compiler                  gc
Test case                 fp
------------------------: O0
build command: gcc -o t-float-libcall -fno-inline -fno-builtin -fno-ipa-cp -fno-omit-frame-pointer -D_FORTIFY_SOURCE=0  -fpatchable-function-entry=5 -O0  s-float-libcall.c   -lm 
test command: /home/honggyu/work/uftrace/uftrace live --no-pager --no-event --libmcount-path=/home/honggyu/work/uftrace/uftrace -A "expf@fparg1/32" -R "expf@retval/f32" -A "log@fparg1/64"  -R "log@retval/f64"   t-float-libcall
uftrace: /home/honggyu/work/uftrace/cmds/record.c:1657:check_binary
 ERROR: Can't find 'mcount' symbol in the 't-float-libcall'.
        It seems not to be compiled with -pg or -finstrument-functions flag.
        You can rebuild your program with it or use -P option for dynamic tracing.
t198_lib_arg_float: -fpatchable-function-entry=5 -O0 returns 1

198 lib_arg_float       : NZ
```
So the logic has to be enabled when `if self.p_flag != '' and len(self.p_libs) > 0` is true. But I used `self.p_flag == '-P .'` instead of `self.p_flag != ''` to make it more clear and readable.

You can test it by running `./runtest.py lib_`. My change makes all the `fpatchable-fun` tests pass.
```
$ ./runtest.py lib_
Start 10 tests with 6 worker
Compiler                  gcc                                           clang                                       
Test case                 pg             finstrument-fu fpatchable-fun  pg             finstrument-fu fpatchable-fun
------------------------: O0 O1 O2 O3 Os O0 O1 O2 O3 Os O0 O1 O2 O3 Os  O0 O1 O2 O3 Os O0 O1 O2 O3 Os O0 O1 O2 O3 Os
073 lib_filter          : OK OK OK OK OK OK OK OK OK OK OK OK OK OK OK  OK OK OK OK OK OK OK OK OK OK OK OK OK OK OK
074 lib_trigger         : OK OK OK OK OK OK OK OK OK OK OK OK OK OK OK  OK OK OK OK OK OK OK OK OK OK OK OK OK OK OK
075 lib_arg             : OK OK OK OK OK SK SK SK SK SK OK OK OK OK OK  OK OK OK OK OK SK SK SK SK SK OK OK OK OK OK
076 lib_replay_F        : OK OK OK OK OK OK OK OK OK OK OK OK OK OK OK  OK OK OK OK OK OK OK OK OK OK OK OK OK OK OK
077 lib_replay_T        : OK OK OK OK OK OK OK OK OK OK OK OK OK OK OK  OK OK OK OK OK OK OK OK OK OK OK OK OK OK OK
131 lib_dlopen          : OK OK OK OK OK OK OK OK OK OK OK OK OK OK OK  OK OK OK OK OK OK OK OK OK OK OK OK OK OK OK
168 lib_nested          : OK OK OK OK OK OK OK OK OK OK OK OK OK OK OK  OK OK OK OK OK OK OK OK OK OK OK OK OK OK OK
192 lib_name            : OK OK OK OK OK OK OK OK OK OK OK OK OK OK OK  OK OK OK OK OK OK OK OK OK OK OK OK OK OK OK
198 lib_arg_float       : OK OK OK OK OK SK SK SK SK SK OK OK OK OK OK  OK OK OK OK OK SK SK SK SK SK OK OK OK OK OK
200 lib_dlopen2         : OK OK OK OK OK NG NG NG NG NG OK OK OK OK OK  OK OK OK OK OK NG NG NG NG NG OK OK OK OK OK
```

Reply:
Hmm.. can we just check the length of `p_libs` then?

Reply:
Ack.